### ISSUE TITLE
Does auth still work for you?

### ISSUE THREAD

**Issue by piec** [knq/hilink#17]
Hi All,

Thanks for this project.

I have a question about auth support. I have a E3372 but auth doesn't seem to work for me.

My setup:
Hardware version: | CL2E3372HM
-- | --
Software version: | 22.328.62.00.1080
Web UI version: | 17.100.19.04.1080

I added my auth details to the main:
```diff
@@ -201,10 +201,12 @@ func main() {
        // hilink options
        opts := []hilink.Option{
                hilink.URL(*flagEndpoint),
+               hilink.Auth("admin", "<my password>"),
        }
```

I see that it tries to login but I'm getting "invalid username or password". I guess the password hashing may have changed? [here](https://github.com/knq/hilink/pull/11/files#diff-f93e21163ef0aae6280cc5e93aa5c072R235-R247)
Or did I do something wrong?
Thanks

---

When I try:
```shell
go run github.com/knq/hilink/cmd/hlcli SmsList -v
```
I'm getting this
```
2020/06/11 17:22:29 POST /api/user/login HTTP/1.1
Host: 192.168.8.1
User-Agent: Go-http-client/1.1
Content-Length: 194
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Cookie: SessionID=/4XvosRBYrmAsm+JEJCjufVTkygUNVRnMQAG3+iALH+I1DN15c9ZBvLosLZ1sLnlry2JPkMoGtjRGLRzpROl4s1XrPeBLXZySDQlmw3MlB/gn5uTzSwUv7B+rJ8kw9Gm
__requestverificationtoken: gnKlbK297/JklGCZf++KnlYeXjLonv1A
Accept-Encoding: gzip

<request>
  <Password>Yjc5NWJhMGYxOTcwNzc3MjQzOWQ1ZDk3YzQ1N2MwZTNkMDk0ZTk5NWFkNDllNDAzNzY4YjUxYjE0MmQ0OTc2OQ</Password>
  <Username>admin</Username>
  <password_type>4</password_type>
</request>
2020/06/11 17:22:29 HTTP/1.1 200 OK
Connection: close
Content-Length: 101
Cache-Control: no-cache
Content-Type: text/html
Date: Thu, 01 Jan 1970 00:00:00 GMT
Server: WebServer
Strict-Transport-Security: max-age=31536000; includeSubdomains
X-Download-Options: noopen
X-Frame-Options: deny
X-Xss-Protection: 1; mode=block
__requestverificationtoken: J2VAtfVmAT2ANp1JlPZBRcapHAeWeiNN

<?xml version="1.0" encoding="UTF-8"?>
<error>
<code>108006</code>
<message></message>
</error>
error: hilink error 108006: invalid username or password
```

**Comment by kenshaw** [comment]
Unfortunately, I don't even have access to hardware to test this with anymore. May I suggest disabling password authentication on the device temporarily? I realize that it's a security issue, but if I remember, access to the management UI is only available on the `192.168/16` subnet.

**Comment by piec** [comment]
Yes I think I'll disable auth for now.
I'll take a look at the client js if I really need it.

Thanks
